src/components/date/date.spec.tsx

```diff
@@ -263,8 +385,8 @@ describe("Date", () => {
   });
 
   describe("when the blur event is triggered on the input", () => {
-    let onBlurFn;
-    let onChangeFn;
+    let onBlurFn: jest.Mock | null;
```

Comment:
```suggestion
    let onBlurFn: jest.Mock | undefined;
```

as in an earlier comment, this means that the `|| jest.fn` can be removed (line 466).

Sadly this doesn't seem possible for the `onChange` due to that not being an optional prop.

Reply:
Actually it looks like nothing is needed here, just `: jest.Mock;` appears to be fine